Rippleitinnz-bloom-filtering-patch1 [DO NOT MERGE]

rippleitinnz opened the pull request:
Moving deduplication to bloom filtering.

Key changes:

Created a bloom_filter class that uses 32MB of memory
Added a typedef using rollover_hashset = bloom_filter; for compatibility
Created global instances for recent_peermsg_hashes and recent_selfmsg_hashes
All files now include bloom_filter.hpp instead of rollover_hashset.hpp
Removed local declarations of the hash sets since they're now global

This maintains full compatibility with the existing code while switching to the bloom filter implementation.

rippleitinnz committed:
Merge pull request #1 from rippleitinnz/rippleitinnz-patch-1

Increase MAX_IN_MSG_QUEUE

rippleitinnz committed:
Merge pull request #3 from rippleitinnz/rippleitinnz-patch-3

Increase MAX_QUEUE_SIZE size to handle larger UNL

rippleitinnz committed:
Merge pull request #2 from rippleitinnz/rippleitinnz-patch-2

increase MAX_NPL_MSG_QUEUE_SIZE and MAX_CONTROL_MSG_QUEUE_SIZE

rippleitinnz committed:
Merge branch 'EvernodeXRPL:main' into main

rippleitinnz committed:
Update rollover_hashset.cpp

Updating deduplication to bloom filtering

rippleitinnz committed:
Update rollover_hashset.hpp

moving deduplication to bloom filters

rippleitinnz committed:
Update rollover_hashset.hpp

Moving deduplication to bloom filters

rippleitinnz committed:
Create bloom_filter.hpp

moving deduplication to bloom filters

rippleitinnz committed:
Update peer_comm_session.cpp

moving deduplication to bloom filters

rippleitinnz committed:
Update CMakeLists.txt

Moving deduplication to bloom filters

rippleitinnz committed:
Update self_node.cpp

moving deduplication to bloom filters

rippleitinnz committed:
Update consensus.cpp

moving deduplication to bloom filters

rippleitinnz committed:
Update usr.hpp

moving deduplication to bloom filters

rippleitinnz commented:
Moving deduplication to bloom filtering.

Summary
Key changes:

Created a bloom_filter class that uses 32MB of memory
Added a typedef using rollover_hashset = bloom_filter; for compatibility
Created global instances for recent_peermsg_hashes and recent_selfmsg_hashes
All files now include bloom_filter.hpp instead of rollover_hashset.hpp
Removed local declarations of the hash sets since they're now global

This maintains full compatibility with the existing code while switching to the bloom filter implementation.

RichardAH commented on src/util/rollover_hashset.cpp:
```diff
@@ -1,3 +1,5 @@
+#**DEPRECATED:** This file is no longer required as deduplication is handled by bloom filteringand will be removed soon. Please do not use.
```
just git rm remove the files

RichardAH commented on src/p2p/peer_comm_session.cpp:
```diff
@@ -13,10 +13,7 @@ namespace p2pmsg = msg::fbuf::p2pmsg;
-    util::rollover_hashset recent_peermsg_hashes(200);
-
-    /**
+     /**
```
where's the actual use of the bloom filter to deduplicate? you probably need to run two at the same time so that periodically you can clear one or the other to prevent them filling up and becoming useless

rippleitinnz replied:
The bloom filter implementation maintains the exact same interface (try_emplace) so no other code changes are needed. The global recent_peermsg_hashes is now defined in bloom_filter.hpp and works exactly the same way.

Will look at running two bloom filters at the same time and will amend

rippleitinnz committed:
Delete src/util/rollover_hashset.cpp

rippleitinnz committed:
Delete src/util/rollover_hashset.hpp

rippleitinnz committed:
Update bloom_filter.hpp

Added in rolling bloom filter.
Two 16MB filters (32MB total) that rotate every 5 minutes
When checking, we look in both filters. When inserting, we add to both.
Every 5 minutes, the older filter is cleared and becomes the new filter
Since we insert into both filters, messages are retained for 5-10 minutes
Uses atomic operations to ensure only one thread performs rotation

The behaviour is:

Minutes 0-5: Filter 1 is active, Filter 2 is building
Minute 5: Clear Filter 1, switch to Filter 2 as active
Minutes 5-10: Filter 2 is active, Filter 1 is building
Minute 10: Clear Filter 2, switch to Filter 1 as active
and so on...
